components/brave_news/rust/lib.rs

```diff
@@ -26,10 +27,64 @@ mod ffi {
     }
 
     extern "Rust" {
+        fn strip_html(subject: &str) -> String;
         fn parse_feed_bytes(source: &[u8], output: &mut FeedData) -> bool;
     }
 }
 
+// Note: This function isn't likely to be perfect, but it does guarantee that
+// there will be no HTML tags in the output (as it strips both '<' and '>' from
+// from the output).
+fn strip_html(subject: &str) -> String {
+    let mut output = String::with_capacity(subject.len());
+    let mut depth_tag = 0;
+    let mut depth_comment = 0;
+
+    // We want to know the last 3 characters so we can see if they were --> for
+    // closing a comment.
+    let last_chars_size = 3;
+    let mut last_chars = String::with_capacity(last_chars_size);
+    let mut iterator = subject.chars();
+
+    while let Some(c) = iterator.next() {
+        last_chars.push(c);
+        if last_chars.len() > last_chars_size {
+            last_chars = last_chars.chars().skip(1).collect();
+        }
+
+        if c == '<' {
+            let is_comment = iterator.clone().take(3).collect::<String>() == "!--";
```

Comment:
I think this can be
```rust
let is_comment = iterator.as_str().get(..3).map_or(false, |s| s == "!--");
```
to avoid the allocation. However, I can replace this with `let is_comment = false` and the tests still pass, so maybe add coverage before taking that suggestion. :)